classes/ocis_manager.php

```diff
@@ -0,0 +1,496 @@
+                    'unauthorized_error',
+                    'repository_ocis',
+                    '',
+                    null,
+                    $e->getTraceAsString()
+                );
+            } catch (InternalServerErrorException $e) {
+                throw new moodle_exception(
+                    'internal_server_error',
+                    'repository_ocis',
+                    '',
+                    null,
+                    $e->getTraceAsString()
+                );
+            }
+        }
+        return $this->drive;
+    }
+
+    /**
+     * @return array<Drive>
+     * @throws BadRequestException
+     * @throws ForbiddenException
+     * @throws InvalidResponseException
+     * @throws NotFoundException
+     * @throws moodle_exception
+     */
+    private function get_drives(): array {
+        $drives = [];
+        try {
+            if ($this->showpersonaldrive) {
+                $drives = $this->get_ocis_client()->getMyDrives(
+                    DriveOrder::NAME,
+                    OrderDirection::ASC,
+                    DriveType::PERSONAL
+                );
+            }
+            if ($this->showshares) {
+                $drives = array_merge(
+                    $drives,
+                    $this->get_ocis_client()->getMyDrives(
+                        DriveOrder::NAME,
+                        OrderDirection::ASC,
+                        DriveType::VIRTUAL
+                    )
+                );
+            }
+            if ($this->showprojectdrives) {
+                $drives = array_merge(
+                    $drives,
+                    $this->get_ocis_client()->getMyDrives(
+                        DriveOrder::NAME,
+                        OrderDirection::ASC,
+                        DriveType::PROJECT
+                    )
+                );
+            }
+        } catch (HttpException $e) {
+            throw new moodle_exception(
+                'could_not_connect_error',
+                'repository_ocis',
+                '',
+                null,
+                $e->getTraceAsString()
+            );
+        } catch (UnauthorizedException $e) {
+            throw new moodle_exception(
+                'unauthorized_error',
+                'repository_ocis',
+                '',
+                null,
+                $e->getTraceAsString()
+            );
+        } catch (InternalServerErrorException $e) {
+            throw new moodle_exception(
+                'internal_server_error',
+                'repository_ocis',
+                '',
+                null,
+                $e->getTraceAsString()
+            );
+        }
+
+        if (empty($drives)) {
+            throw new \moodle_exception(
+                'no_drives_error',
+                'repository_ocis',
+                '',
+                null
+            );
+        }
+        return $drives;
+    }
+
+    private function get_drive_listitem(Drive $drive): array|null {
```

Comment:
```suggestion
    private function get_drive_list_item(Drive $drive): array|null {
```

There is `get_resource_list_item` - so maybe make `get_drive_list_item` have the same format.